internal/services/compute/capacity_reservation_resource.go

```diff
@@ -0,0 +1,282 @@
+		Create: resourceCapacityReservationCreate,
+		Read:   resourceCapacityReservationRead,
+		Update: resourceCapacityReservationUpdate,
+		Delete: resourceCapacityReservationDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.CapacityReservationID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"capacity_reservation_group_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"zone": commonschema.ZoneSingleOptionalForceNew(),
+
+			"sku": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"capacity": {
+							Type:         pluginsdk.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntAtLeast(1),
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceCapacityReservationCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.CapacityReservationsClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	capacityReservationGroupName := d.Get("capacity_reservation_group_name").(string)
+	id := parse.NewCapacityReservationID(subscriptionId, resourceGroup, capacityReservationGroupName, name)
+	existing, err := client.Get(ctx, id.ResourceGroup, id.CapacityReservationGroupName, id.Name, "")
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for existing %s: %+v", id, err)
+		}
+	}
+	if !utils.ResponseWasNotFound(existing.Response) {
+		return tf.ImportAsExistsError("azurerm_capacity_reservation", id.ID())
+	}
+
+	parameters := compute.CapacityReservation{
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Sku:      expandCapacityReservationSku(d.Get("sku").([]interface{})),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if v, ok := d.GetOk("zone"); ok {
+		zones := []string{v.(string)}
+		parameters.Zones = &zones
+	}
```

Comment:
The API exposed `zones`, however, it can only accept one `zone`, so used the non-common way to expand/flatten this. Similar to `managed_disk.zone` https://github.com/hashicorp/terraform-provider-azurerm/blob/7c27c458db592eb8ffc47b2fe646f0b00f399f48/internal/services/compute/managed_disk_resource.go#L472-L476